### PR TITLE
dev/core/17 Convert interview task to use Pseudoconstants

### PR DESCRIPTION
**Pull request by mattwire**
Overview
----------------------------------------
Remove deprecated CRM_Core_OptionGroup and replace with CRM_Core_PseudoConstant.  Fix PHP notices.

Ref https://lab.civicrm.org/dev/core/issues/17

@eileenmcnaughton One of your favourites?

### CODE REVIEW

File: CRM/Campaign/Form/Task/Interview.php [hunk 1]
```diff
@@ -580,10 +581,10 @@ public function getVoterIds() {
     $this->_contactIds = $this->get('contactIds');
     if (!is_array($this->_contactIds)) {
       //get the survey activities.
-      $activityStatus = CRM_Core_PseudoConstant::activityStatus('name');
+      $scheduledStatusId = CRM_Core_PseudoConstant::getKey('CRM_Activity_DAO_Activity', 'activity_status_id', 'Scheduled');
       $statusIds = array();
-      if ($statusId = array_search('Scheduled', $activityStatus)) {
-        $statusIds[] = $statusId;
+      if ($scheduledStatusId) {
```
Comment (eileenmcnaughton) [inline CRM/Campaign/Form/Task/Interview.php]: should this not be if ($scheduledStatusId === $statusId) {

**Comment by eileenmcnaughton** [comment]
These are pretty straight forward changes & I read through them & had only one concern (commented in line)

File: CRM/Campaign/Form/Task/Interview.php (hunk 1 above)
Comment (mattwire) [reply CRM/Campaign/Form/Task/Interview.php]: @eileenmcnaughton No, the original code was checking if $scheduledStatusId had a value so I just updated it.  However, looking again, there is scope for simplifying the code a little which removes that `if` entirely.  Does it look ok now?

**Comment by eileenmcnaughton** [comment]
Test fail is unrelated & I think the change makes sense now. I would note that if the status 'Scheduled' does not exist it might be a bit unpredictable - but I think we DO expect that to exist.

**Pull request merged**